include/parallel/threads.h

```diff
@@ -61,6 +61,25 @@ class BoolAcquire
   bool & _b;
 };
 
+/**
+ * We use a class to turn perf logging off and on within threads, to
+ * be exception-safe and to avoid forcing indirect inclusion of
+ * libmesh_logging.h everywhere.
+ */
+class DisablePerfLogInScope
+{
+public:
+#ifndef LIBMESH_ENABLE_PERFORMANCE_LOGGING
+  DisablePerfLogInScope() {}
+  ~DisablePerfLogInScope() {}
```

Comment:
`= default;` for the "non logging" implementations rather than empty brackets? 

The [reasoning](https://stackoverflow.com/questions/13576055/how-is-default-different-from-for-default-constructor-and-destructor) is related to whether the compiler can consider it a "trivial" class or not.

Reply:
Oh, man, I'd forgotten about that.  Will fix.